guvnor-ala/guvnor-ala-openshift/guvnor-ala-openshift-ui/guvnor-ala-openshift-ui-client/src/main/java/org/guvnor/ala/ui/openshift/client/pipeline/template/TemplateParamsFormPresenter.java

```diff
@@ -461,10 +466,20 @@ private void addTemplateParam(StringBuilder builder,
         builder.append(paramValue);
     }
 
-    private String buildContainerParamValue(final String container,
-                                            final GAV gav) {
-        return container + "=" + gav.getGroupId() +
-                ":" + gav.getArtifactId() +
-                ":" + gav.getVersion();
+    protected String buildContainerDeploymentParamValue(final List<ContainerConfig> containerConfigs) {
+        StringBuilder builder = new StringBuilder();
+        containerConfigs.forEach(containerConfig -> {
+            if (builder.length() > 0) {
+                builder.append(CONTAINER_DELIMITER);
+            }
+            builder.append(buildContainerParamValue(containerConfig));
+        });
+        return builder.toString();
```

Comment:
A proper data object would be more safe to rely on, instead of a String with pipes....

With that said.. I also don't see any major issue.. so I'd leave as is for now.

Reply:
Agreed. If we do change to some other format for this value (like JSON, maybe), then we'd have to also make a change to support it in the s2i scripts on the openshift side, which as of right now only support the pipe-delimited form.

Aside, I'm curious if this project has access to the org.kie.server.services.impl.KieServerContainerDeployment class, which can build this (current) format now for you instead of duplicating it here. You can see it's usage in org.kie.server.services.impl.KieServerContainerDeploymentTest.

Reply:
@errantepiphany the KieServerContainerDeployment class is not accesible from GWT client side code. To use it we could eventually:
1) make kie-server-services-common module to be also a GWT module, but this leads us on selecting which subset of classes are GWT complatible, and also keep them compatible in the future, etc.

2) add a service method like String makeTheContainerParamStringValue(the params) and inside the server side implementation we can access KieServerContainterDeployment

but to be honest I'd prefer this small duplication by now.

Reply:
@wmedvede That makes sense, and I agree with your preference.